backend/compact-connect/stacks/persistent_stack/compact_configuration_upload.py

```diff
@@ -0,0 +1,163 @@
+                              'and is scoped to one table and encryption key.'
+                }
+            ]
+        )
+
+
+        self.compact_configuration_upload_provider = Provider(
+            scope, 'CompactConfigurationUploadProvider',
+            on_event_handler=self.compact_configuration_upload_function,
+            log_retention=RetentionDays.ONE_DAY
+        )
+        NagSuppressions.add_resource_suppressions_by_path(
+            Stack.of(scope),
+            f'{self.compact_configuration_upload_provider.node.path}/framework-onEvent/Resource', [
+                {
+                    'id': 'AwsSolutions-L1',
+                    'reason': 'We do not control this runtime'
+                },
+                {
+                    'id': 'HIPAA.Security-LambdaConcurrency',
+                    'reason': 'This function is only run at deploy time, by CloudFormation and has no need for '
+                              'concurrency limits.'
+                },
+                {
+                    'id': 'HIPAA.Security-LambdaDLQ',
+                    'reason': 'This is a synchronous function run at deploy time. It does not need a DLQ'
+                },
+                {
+                    'id': 'HIPAA.Security-LambdaInsideVPC',
+                    'reason': 'We may choose to move our lambdas into private VPC subnets in a future enhancement'
+                },
+            ]
+        )
+
+        NagSuppressions.add_resource_suppressions_by_path(
+            Stack.of(scope),
+            path=f'{self.compact_configuration_upload_provider.node.path}'
+                 f'/framework-onEvent/ServiceRole/DefaultPolicy/Resource',
+            suppressions=[
+                {
+                    'id': 'AwsSolutions-IAM5',
+                    'reason': 'The actions in this policy are specifically what this lambda needs to read '
+                              'and is scoped to one table and encryption key.'
+                }
+            ]
+        )
+
+        NagSuppressions.add_resource_suppressions_by_path(
+            Stack.of(scope),
+            path=f'{self.compact_configuration_upload_provider.node.path}/framework-onEvent/ServiceRole/Resource',
+            suppressions=[
+                {
+                    'id': 'AwsSolutions-IAM4',
+                    'reason': 'This role is managed by AWS CDK.'
+                }
+            ]
+        )
+
+        self.compact_configuration_uploader_custom_resource = CustomResource(
+            scope, 'CompactConfigurationUploadCustomResource',
+            resource_type='Custom::CompactConfigurationUpload',
+            service_token=self.compact_configuration_upload_provider.service_token,
+            properties={
+                # passing this as a property to the custom resource so that the lambda can access it
+                'compact_configuration': self._generate_compact_configuration_json_string(),
```

Comment:
Rather than passing in the environment name and whether it is a sandbox, it may make sense to just filter out non-active configurations here at Synth-time, rather than leave it to the custom resource to ignore irrelevant configs.